### PR TITLE
MULE-12255: Migrate testing namespaces to plugins

### PR DESCRIPTION
**Pull request by elrodro83**
* move unit tests of testing components back to its own module
* move compatibility specific stuff to compatibility module
* move connector specific stuff to the corresponding connector
* remove dead code

### CODE REVIEW

File: tests/component-plugin/src/main/java/org/mule/functional/api/component/FunctionalTestComponent.java [hunk 1]
```diff
@@ -210,21 +182,20 @@ protected String append(String contents, Event event) {
...
-      messageHistory.add(data);
+      messageHistory.add(event);
     }
 
+    final Message message = event.getMessage();
     if (logger.isInfoEnabled()) {
-      String msg = StringMessageUtils.getBoilerPlate("Message Received in service: "
-          + context.getFlowConstruct().getName() + ". Content is: "
-          + StringMessageUtils.truncate(data.toString(), 100, true), '*', 80);
+      String msg = getBoilerPlate("Message Received in service: "
```
Comment (pablokraan) [inline tests/component-plugin/src/main/java/org/mule/functional/api/component/FunctionalTestComponent.java]: service -> flow?

File: tests/component-plugin/src/main/java/org/mule/functional/api/notification/FunctionalTestNotification.java [hunk 2]
```diff
@@ -40,22 +42,22 @@
...
+  private final Event event;
 
-  public FunctionalTestNotification(MuleEventContext context, Object replyMessage, int action)
+  public FunctionalTestNotification(Event event, FlowConstruct flowConstruct, Object replyMessage, int action)
       throws TransformerException {
-    super(context.getMessage().getPayload().getValue(), action);
+    super(event.getMessage().getPayload().getValue(), action);
     // TODO(pablo.kraan): MULE-12104: flow construct should not be null. Remove the workaround when fixed
-    resourceIdentifier = context.getFlowConstruct() == null ? "UNKNOWN" : context.getFlowConstruct().getName();
+    resourceIdentifier = flowConstruct == null ? "UNKNOWN" : flowConstruct.getName();
```
Comment (pablokraan) [inline tests/component-plugin/src/main/java/org/mule/functional/api/notification/FunctionalTestNotification.java]: Can you check if the issue mentioned in the TODO is fixed with your changes?

File: tests/component-plugin/src/main/resources/META-INF/mule-artifact/mule-plugin.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "classLoaderModelLoaderDescriptor": {
+    "id": "maven",
+    "attributes": {
+      "exportedPackages": [
+        "org.mule.functional.api.component",
+        "org.mule.functional.api.exception",
+        "org.mule.functional.api.notification"
+      ],
+      "exportedResources": [
+        "META-INF/mule-test.xsd"
```
Comment (pablokraan) [inline tests/component-plugin/src/main/resources/META-INF/mule-artifact/mule-plugin.json]: Is still needed to export the XSD?

File: tests/component-plugin/src/test/java/org/mule/test/tck/AssertionMessageProcessorTestCase.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  }
+
+  @Test
+  public void processValidEvent() throws Exception {
+    AssertionMessageProcessor asp = createAssertionMessageProcessor();
+    asp.setFlowConstruct(flowConstruct);
+    asp.setExpression(TRUE_EXPRESSION);
+    asp.start();
+    asp.process(mockEvent);
+    assertFalse(asp.expressionFailed());
```
Comment (pablokraan) [inline tests/component-plugin/src/test/java/org/mule/test/tck/AssertionMessageProcessorTestCase.java]: Use assertThat

File: tests/functional/src/main/java/org/mule/functional/transformer/ValidateResponse.java [hunk 5]
```diff
@@ -31,7 +32,7 @@ protected Object doTransform(Object src, Charset encoding) throws TransformerExc
     if (response != null && response.contains("success")) {
       return response;
     } else {
-      throw new TransformerException(I18nMessageFactory.createStaticMessage("Invalid response from service: " + response));
+      throw new TransformerException(createStaticMessage("Invalid response from service: " + response));
```
Comment (pablokraan) [inline tests/functional/src/main/java/org/mule/functional/transformer/ValidateResponse.java]: service -> flow?

File: tests/test-policies/src/test/java/org/mule/test/policy/PolicyTestCase.java [hunk 6]
```diff
@@ -35,7 +35,8 @@
 
 import org.junit.Test;
 
-public class PolicyTestCase extends MuleArtifactFunctionalTestCase {
+//TODO MULE-12760 Make this test extend MuleArtifactFunctionalTestCase
+public class PolicyTestCase extends FunctionalTestCase {
```
Comment (pablokraan) [inline tests/test-policies/src/test/java/org/mule/test/policy/PolicyTestCase.java]: Why is this required?

Comment (elrodro83) [inline tests/test-policies/src/test/java/org/mule/test/policy/PolicyTestCase.java]: There seems to be some issue with the runner for this case of having tests in the same module as the plugin

File: tests/component-plugin/src/main/resources/META-INF/mule-artifact/mule-plugin.json (hunk 3 above)
Comment (elrodro83) [inline tests/component-plugin/src/main/resources/META-INF/mule-artifact/mule-plugin.json]: yes, even the extensions generator is exporting the schema of each extension

File: modules/deployment/src/test/resources/META-INF/mule-module.properties [hunk 7]
```diff
@@ -1,3 +1,6 @@
-    module.name=deployment-test
+module.name=deployment-test
 
-artifact.export.classPackages=org.mule.runtime.module.deployment.internal
+artifact.export.classPackages=org.mule.runtime.module.deployment.internal,\
+                              org.mule.functional.policy
```
Comment (elrodro83) [inline modules/deployment/src/test/resources/META-INF/mule-module.properties]: the way the deployment tests run in a non-isolated manner, so the config from the plugin is not used. Anyway, i don't need to export this package here, but i still need to export the xsd.
